File: Java/Desktop/ReadyToExam/src/NewYearGift/Model/Entity/Chocolate.java
package NewYearGift.Model.Entity;

import NewYearGift.Model.Entity.Abstract.Candy;

public class Chocolate extends Candy{


    private String type;
    private boolean withFilling;
    private String form;


    public Chocolate(double weight, String creator,double price, String type, boolean withFilling, String form) {
        super(weight, creator,price);
        this.type = type;
        this.withFilling = withFilling;
        this.form = form;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public boolean isWithFilling() {
        return withFilling;
    }

    public void setWithFilling(boolean withFilling) {
        this.withFilling = withFilling;
    }

    public String getForm() {
        return form;
    }

    public void setForm(String form) {
        this.form = form;
    }

    @Override
    public String toString() {
        return new StringBuilder().append("Chocolate ").append(super.toString()).append("Type: ").append(type).append("\nWith filling? - ").append(withFilling).append("\nForm: ").append(form).toString();
    }
}

File: Csharp/ReadyToExam/ReadyToExam/NewYearGift/Model/Entity/Chocolate.cs
using System.Text;
using ReadyToExam.NewYearGift.Model.Abstracts;

namespace ReadyToExam.NewYearGift.Model.Entity
{
    public class Chocolate : Candy
    {
        private string _type;
        private bool _withFilling;
        private string _form;

        public Chocolate(double weight, string creator, double price, string type, bool withFilling, string form) : base(weight, creator, price)
        {
            _type = type;
            _withFilling = withFilling;
            _form = form;
        }

        public string Type
        {
            get => _type;
            set => _type = value;
        }

        public bool WithFilling
        {
            get => _withFilling;
            set => _withFilling = value;
        }

        public string Form
        {
            get => _form;
            set => _form = value;
        }

        public override string ToString()
        {
            return new StringBuilder().Append("Chocolate " + base.ToString() + "\nType = " + _type +
                                              "\nWith Filling or not? - " + _withFilling + "\nForm = " + _form)
                .ToString();
        }
    }
}